### ISSUE TITLE
Cannot init libpython3.5.so from dotnet core 2.0 in Ubuntu

### ISSUE THREAD

**Issue by karpovilia** [pythonnet/pythonnet#565]
### Environment

-   Pythonnet version: latest master from github
-   Python version: 3.5 64-bit
-   Operating System:  Ubuntu 16.04

### Details

I'm trying to run simple c# test app (x64), which calls python "helloworld" script.
The code works perfectly in Windows Server 2008, but doesn't work in Ubuntu 16.04
I've added some Console.Writeline outputs. The library crushes in PythonEngine.Initialize() method.
    
```
dotnet PythonTest.dll
trying to load LoadLibrary + <<</usr/lib/python3.5/config-3.5m-x86_64-linux-gnu/libpython3.5.so>>>
dlopen /usr/lib/python3.5/config-3.5m-x86_64-linux-gnu/libpython3.5.so
resint = 0
Calling GetProcAddress, with parameters dllHandle = 0, and name = _PyObject_NextNotImplemented
look in the exe if dllHandle is NULL
Segmentation fault (core dumped)
```

Maybe I've used wrong conditional compilation symbols or RTLD_GLOBAL/RTLD_SHARED constants. Can you please write a list of conditional compilation symbols to compile the library for dotnet core 2.0 in Ubuntu or publish compiled library?

**Comment by denfromufa** [comment]
How did you install and why you cannot use Mono?

**Comment by dmitriyse** [comment]
The problem relates to PYTHON_WITH_PYMALLOC constant.PYTHON_WITH_PYMALLOC
Python.Runtime.dll should be compiled with PYTHON_WITH_PYMALLOC for libpython3.5m.so
And without this constant for libpython3.5.so

Currently under linux Python.Net was debugged only with libpython3.5m.so libraries

**Comment by dmitriyse** [comment]
Please try to build Python.Net from scratch solely under linux. Probably some Windows builded files was used (from shared folder for example)

**Comment by denfromufa** [comment]
@dmitriyse doesn't setup.py detect PYTHON_WITH_PYMALLOC during installation?

**Comment by dmitriyse** [comment]
I never faced in my practiace with the case when PYTHON_WITH_PYMALLOC was not detected.
Logic of PYTHON_WITH_PYMALLOC detection is exactly the same for the mono build and the xplat build

**Comment by filmor** [comment]
Closed in favour of #984, we will pass the library name more dynamically in the future which will resolve issues like this and similar problems.
